src/plugins/identity/alias.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import {type NodeAddressT} from "../../core/graph";
+import {loginAddress as githubAddress} from "../github/nodes";
+import {userAddress as discourseAddress} from "../discourse/createGraph";
+
+/** An Alias is a string specification of an identity within another plugin.
+ *
+ * For now, the supported alias forms are `github/${githubUsername}` and
+ * `discourse/${discourseUsername}`. As a courtesty, if the user has put an @
+ * before the username, we strip it for them.
+ *
+ * This format has been chosen to be moderately extensible and easy to maintain
+ * by hand, as in the near future the alias files will be maintained by hand.
+ * This system will not scale well when user-provided plugins need to add to the
+ * aliasing scheme, so at that point we will rewrite this system.
+ */
+export type Alias = string;
+
+export function resolveAlias(
+  alias: Alias,
+  discourseUrl: string | null
```

Comment:
This API (along with the `githubAddress` and `discourseAddress` dependencies) isn't very extensible or neutral as a plugin. Instead, you might provide a map like `Map<string, function>` where string is your type/prefix, the function has a `UsernameT => NodeAddressT` signature.

Reply:
You might even consider turning this into a class or factory, to separate providing the functions from resolving an alias.

```
// factory style
const createAliasResolver = (mapping: Map<string, function>) => (alias: Alias) => { ... }

// OO style
class AliasResolver {
 ...
 constructor (mapping: Map<string, function>) { ... }
 resolve (alias: Alias) { ... }
}
```

Reply:
This is a very reasonable suggestion. However, I'm not going to do so in this PR. Right now, the Discourse and GitHub plugins are explicitly privileged all over the place (e.g. the project data format explicitly lists repo ids and a discourse server, the CLI explicitly looks for these keys, etc). I don't think there's much benefit to modularizing the plugin system piecemeal; instead, I'm planning to do a big refactor later, when we start to see external interest in adding plugins. At that point we can ensure the entire system has a clean and coherent design.